insights/C/generate/generate_linux.go

```diff
@@ -28,43 +31,30 @@ func buildSharedLibs() error {
 		constants.Version = version
 	}
 
-	re := regexp.MustCompile(`(\d+)\.(\d+)\.(\d+)`)
-	matches := re.FindStringSubmatch(constants.Version)
-
-	libname := "libinsights.so"
-	if len(matches) == 4 {
-		libname = fmt.Sprintf("libinsights.so.%s", matches[1])
-	}
-
 	ldflags := fmt.Sprintf("-X=constants.Version/internal/constants.Version=%s -extldflags -Wl,-soname,%s", constants.Version, libname)
 
-	cmd := exec.Command("go", "build", //nolint:gosec // This is controlled by the build process and also filtered here
+	if _, err := exec.Command("go", "build", //nolint:gosec // This is controlled by the build process and also filtered here
 		"-buildmode=c-shared",
 		"-trimpath",
 		"-ldflags", ldflags,
 		"-o", fmt.Sprintf("../generated/%s", libname),
 		"./libinsights.go",
-	)
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		slog.Error("Failed to build shared libraries", "stdout", string(output), "error", err)
+	).CombinedOutput(); err != nil {
 		return err
 	}
 
 	// Rename the header file to insights.h
 	lastDot := strings.LastIndex(libname, ".")
 	expectedHeader := libname[:lastDot] + ".h"
 	if err := os.Rename(fmt.Sprintf("../generated/%s", expectedHeader), "../generated/insights.h"); err != nil {
-		slog.Error("Failed to rename header file", "error", err)
+		return err
 	}
-	return err
+	return nil
 }
 
 func copyTypesHeader() error {
-	cmd := exec.Command("cp", "./insights_types.h", "../generated/types.h")
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		slog.Error("Failed to copy types header", "stdout", string(output), "error", err)
+	if _, err := exec.Command("cp", "./insights_types.h", "../generated/types.h").CombinedOutput(); err != nil {
+		return err
```

Comment:
If you dont’ want to return the output here, you can use `Run()` or just augment the error with the output.